### PR TITLE
Added parts token unpledging & Fix : Pledge token sending

### LINKED ISSUE #75: Parts token are not getting unpledged.
Parts tokens pledged for a transaction are not getting unpledged after a successful transaction.

### LINKED ISSUE #152: Pledged token unable to transfer.
When a Quorum become a sender node, unable to transfer the tokens in the node which are already pledged for another and got unpledged.

### PR DESCRIPTION
**Pull request by vaishnav-ch**
- Added parts token type in unpledging and proof verification functions.
- Updated token details fetching method in proof verification function for fixing `leveldb not found` issue during pledged token transfer.

### CODE REVIEW

File: core/quorum_recv.go [hunk 1]
```diff
@@ -209,6 +209,18 @@ func (c *Core) quorumRBTConsensus(req *ensweb.Request, did string, qdc didcrypto
...
+			tknType = token.RBTTokenType
+		} else {
+			tknType = token.PartTokenType
+		}
+
+		if c.testNet && wt[i].TokenValue == 1.0 {
+			tknType = token.TestTokenType
+		} else {
+			tknType = token.TestPartTokenType
+		}
```
Comment (KiranHRubix) [inline core/quorum_recv.go:223]: @vaishnav-ch the token type for the token that is is selected in the ith iteration can be fetched using wt[i].TokenType . 

So there is no logical need for this if conditions 

try changing the code and test

File: core/quorum_recv.go [hunk 2]
```diff
@@ -1280,8 +1292,9 @@ func (c *Core) getProofverificationDetails(tokenID string, senderAddr string) (s
 			return "", "", err
 		}
 		//check if token chain of token pledged for already synced to node
+		//TODO: Change proof verification method
 		pledgedfBlk, err := c.w.GetTokenBlock(tokenPledgedFor, tokenPledgedForType, tokenPledgedForBlockId)
-		if err != nil {
+		if err != nil && err.Error() != "leveldb: not found" {
```
Comment (KiranHRubix) [inline core/quorum_recv.go:1297]: Please add a comment here explaining what the if condition is doing

**Review by KiranHRubix — COMMENTED** [review]
@vaishnav-ch , Please look in to the comments and add the changes.

Also please provide in the PR description the scenarios you have tested, Once these details have been added lets review the PR

File: core/quorum_recv.go (hunk 1 above)
Comment (vaishnav-ch) [reply core/quorum_recv.go]: Removed `tknType` now fetching using `wt[i].TokenType`

File: core/quorum_recv.go (hunk 2 above)
Comment (vaishnav-ch) [reply core/quorum_recv.go]: Condition changed

**Review by KiranHRubix — COMMENTED** [review]
Changes looks good.Please continue with testing and update

**Comment by vaishnav-ch** [comment]
*Test cases*

**Unpledging of parts token**

Node where unpledged parts are existing- Unpledging of parts/whole is working
Node where new parts are pledged- Unpledging of parts/whole is working

**Transfer of Unpledged tokens**

Sender/Receiver- A, B, C, D
Quorums- Q1- Q10

***Case1:  Quorum node to become Sender/Receiver Node***

- A to B & B to A transferred 3 RBTs as multiple transfers with only parts, only whole and combi of parts and whole with Q1- Q5 as quorum. - Transfer working
- Q1 is changed to sender/receiver
- Q1 to C & C to Q1 transferred 2 RBTs (All Unpledged tokens) as multiple transfers with only parts, only whole and combi of parts and whole. - Transfer working

***Case2: Quorum node to become Sender/Receiver Node and again back to be a Quorum Node again***

- A to B & B to A transferred 3 RBTs as multiple transfers with only parts, only whole and combi of parts and whole with Q1- Q5 as quorum. - Transfer working
- Q1 is changed to sender/receiver
- Q1 to C & C to Q1 transferred 2 RBT (All Unpledged tokens) as multiple transfers with only parts, only whole and combi of parts and whole with Q2 to Q6 as quorums- Transfer working
- Q1 is changed to  quorum again.
- A to B and B to A transferred 3 RBTs as multiple transfers with only parts, only whole and combi of parts and whole with Q1- Q5- Transfer working

***Case3: Quorum node to become Sender/Receiver node with a new set of quorum***

- A to B & B to A transferred 3 RBTs as multiple transfers with only parts, only whole and combi of parts and whole with Q1- Q5 as quorum. - Transfer working
- Q1 is changed to sender/receiver
- Q1 to C & C to Q1 transferred 2 RBT (All Unpledged tokens) as multiple transfers with only parts, only whole and combi of parts and whole with Q6 to Q10 as quorums- Transfer working

***Case4: A, B, C, D and Q1 to Q10 become Sender/Receiver/Quorum in random order***
- All transfers working


(will modify after another round of testing)

**Review by KiranHRubix — APPROVED** [review]

**Pull request merged**